ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java

```diff
@@ -98,24 +98,31 @@ public void run() {
             String rightPrompt = String.format("took %s ms", previousDuration.toMillis());
             rightPrompt = terminal.color(rightPrompt, TerminalAdapter.BRIGHT);
 
-            String source = terminal.readLine(leftPrompt, rightPrompt).trim();
-
-            start = Instant.now();
             try {
-                if (!commandHandler.handle(source)) {
-                    String result = evaluator.evaluate(source);
-                    terminal.result(result);
-                }
-            } catch (Exception e) {
-                if (!evaluator.hasErrors()) {
-                    terminal.fatalError("Something went wrong: " + e.getMessage());
+                String source = terminal.readLine(leftPrompt, rightPrompt).trim();
+                start = Instant.now();
+                try {
+                    if (!commandHandler.handle(source)) {
+                        String result = evaluator.evaluate(source);
+                        terminal.result(result);
+                    }
+                } catch (ShellExitException e) {
```

Comment:
Is it an issue if the finally block is executed here? If not, we can have a single try block, right?